woocommerce-gateway-amazon-payments-advanced.php

```diff
@@ -271,6 +271,64 @@ public static function get_site_description() {
 		return self::sanitize_string( $site_description );
 	}
 
+	/**
+	 * Helper method to get order Version.
+	 *
+	 * @return string
+	 */
+	public static function get_order_version( $order_id, $force = true ) {
+		if ( $force && WC_Amazon_Payments_Advanced_Merchant_Onboarding_Handler::get_migration_status() ) {
+			return 'v2';
+		}
+		$order   = wc_get_order( $order_id );
+		$version = version_compare( $order->get_meta( 'amazon_payment_advanced_version' ), '2.0.0' ) >= 0 ? 'v2' : 'v1';
+		return $version;
+	}
+
+	/**
+	 * Helper method to get order Version.
+	 *
+	 * @return string
+	 */
+	public static function get_order_charge_permission( $order_id ) {
+		$order   = wc_get_order( $order_id );
+		$charge_permission_id = $order->get_meta( 'amazon_charge_permission_id' );
+		if ( empty( $charge_permission_id ) ) {
+			// For the subscriptions created on versions previous V2 we update the meta.
+			$charge_permission_id = $order->get_meta( 'amazon_billing_agreement_id' );
+			if ( empty( $charge_permission_id ) ) {
+				// For the orders created on versions previous V2 we update the meta.
+				$charge_permission_id = $order->get_meta( 'amazon_reference_id' );
+			}
+			$order->update_meta_data( 'amazon_charge_permission_id', $charge_permission_id );
+		}
+		return $charge_permission_id;
+	}
+
+	/**
+	 * Helper method to get order Version.
+	 *
+	 * @return string
+	 */
+	public static function get_order_charge_id( $order_id ) {
+		$order   = wc_get_order( $order_id );
+		$charge_id = $order->get_meta( 'amazon_charge_id' );
+		if ( empty( $charge_id ) ) {
+			// For the orders created on versions previous V2 we get the equivalent meta. 
+			$charge_id = $order->get_meta( 'amazon_capture_id');
+			if ( empty( $charge_id ) ) {
+				// For the orders created on versions previous V2 with pending capture 
+				// we adapt the existing meta. 
+				$authorization_id = $order->get_meta( 'amazon_authorization_id' );
+				$charge_id = substr_replace( $authorization_id, 'C', 20, 1 );
```

Comment:
isn't this limited to the size of the ids staying the same? shouldn't we replace something like `-A` to `-C` instead, which is more flexible? Or split, and replace the portion that has the A (the third as I remember) and then join again?

Replacing a specific position in the string feels fragile to me.